### DISCUSSION TITLE
How to debug Cycle errors?

### DISCUSSION THREAD

**Discussion by doczoidberg** [handsontable/hyperformula#539] [General]
we have the problem that we get #Cycle errors which we don't get in Excel. 

It's hard to debug because the cycles seems to be quite huge. Do you have any idea how we could debug this problem?

**Comment by bardek8** [comment]
Well, there is one known issue with cycles: #336 . Is it possible, that you have that kind of logic?
Apart from that you can use the `getCellPrecedents` method for tracking dependent cells.

**Comment by doczoidberg** [comment]
I already use `getCellPrecedents` (which one have to call recursively) but the problem is that I don't know where the invalid reference happens. every cell in the circle has the `#cycle` error

**Comment by izulin** [comment]
You might want to try (still experimental) branch from PR #529 
In the example there you can see how to use it to only show cells that are 'original source' of errors. In your case, this should output all the cells that are on the cycle, ignoring all the cells to which the cycle error is propagated.

**Comment by bardek8** [comment]
And please remember that none of the methods will show you the single one invalid reference. By looking at all cells at the cycle independently, each of them does not cause an error, but only when we have them all.
So one way or the other you'll have to inspect all cells at the cycle.
